### PR TITLE
CARTO: Fix autohighlighting in spatial index tile layers

### PR DESCRIPTION
**Pull request by felixpalmer**
For https://github.com/visgl/deck.gl/issues/6912

#### Change List
- Map spatial index to data index for autoHighlighting

### CODE REVIEW

File: modules/carto/src/layers/spatial-index-tile-layer.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import {TileLayer} from '@deck.gl/geo-layers';
```
Comment (Pessimistress) [inline modules/carto/src/layers/spatial-index-tile-layer.js:1]: Write new classes in TypeScript?

**Review by Pessimistress — APPROVED** [review]

File: modules/carto/src/layers/spatial-index-tile-layer.js (hunk 1 above)
Comment (felixpalmer) [reply modules/carto/src/layers/spatial-index-tile-layer.js:1]: As the other files were in JS in this directory I stuck with that, but the plan is to migrate this next week

Comment (alasarr) [reply modules/carto/src/layers/spatial-index-tile-layer.js:1]: Our plan is to do it right after this PR. I know in the future we should develop directly in typescript, but we were in hurry for this to do QA in other pieces of our platform.

Added to the TODO of spatial indexes: https://github.com/visgl/deck.gl/issues/6912#issuecomment-1129682952 to make sure we don't forget about it

File: modules/carto/src/layers/spatial-index-tile-layer.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  _updateAutoHighlight(info) {
+    const {hoveredFeatureId} = this.state;
+    const hoveredFeature = info.object;
+    let newHoveredFeatureId;
+
+    if (hoveredFeature) {
+      newHoveredFeatureId = hoveredFeature.id;
+    }
+    let {highlightColor} = this.props;
+    if (typeof highlightColor === 'function') {
```
Comment (alasarr) [inline modules/carto/src/layers/spatial-index-tile-layer.js:17]: I feel we can move this piece of code inside of the next if `(hoveredFeatureId !== newHoveredFeatureId)` to save some performance

**Review by alasarr — APPROVED** [review]
Looks good to me, just a small perf thing

**Pull request merged**